treeherder/log_parser/failureline.py

```diff
@@ -180,27 +180,40 @@ def create(job_log, log_list):
 
 def replace_astral(log_list):
     for item in log_list:
-        for key in ["test", "subtest", "message", "stack", "stackwalk_stdout", "stackwalk_stderr"]:
+        for key in [
+            "test",
+            "subtest",
+            "message",
+            "stack",
+            "stackwalk_stdout",
+            "stackwalk_stderr",
+        ]:
             if key in item:
                 item[key] = astral_filter(item[key])
         yield item
 
 
-def get_group_results(repository, push):
-    groups = Group.objects.filter(
-        job_logs__job__push__revision=push.revision,
-        job_logs__job__push__repository=repository,
-        group_result__status__in=[GroupStatus.OK, GroupStatus.ERROR],
-    ).values(
-        "group_result__status",
-        "name",
-        "job_logs__job__taskcluster_metadata__task_id",
+def get_group_results(push):
+    # Use push.id instead of filtering by revision and repository separately
+    # Also use select_related to reduce queries and only fetch needed fields
+    groups = (
+        Group.objects.filter(
+            group_result__job_log__job__push=push,
```

Comment:
what is compared here?  push_id ?

Reply:
also group_result is a serializer- that might introduce more overhead?

Reply:
> what is compared here? push_id ?

Yeah, internally, it compares the `id`.